compiler/passes/convert-uast.cpp

```diff
@@ -4230,20 +4233,34 @@ void ConvertedSymbolsMap::applyFixups(chpl::Context* context,
 
   // Note: we should be able to minimize the fixups needed
   // by converting the modules in the initialization order
+  llvm::SmallPtrSet<SymExpr*, 4> fixedUp;
 
   // Fix up any SymExprs needing to be re-targeted
   for (const auto& p : identFixups) {
     SymExpr* se = p.first;
     ID target = p.second;
 
-    INT_ASSERT(isTemporaryConversionSymbol(se->symbol()));
+    // Already fixed up by following the symExprs linked list on a
+    // TemporaryConversionSymbol (see below). Skip here.
+    if (fixedUp.count(se) > 0) continue;
+
+    auto tcsymbol = se->symbol();
+    INT_ASSERT(isTemporaryConversionSymbol(tcsymbol));
 
     Symbol* sym = findConvertedSym(target, /* trace */ false);
     if (sym == nullptr) {
       INT_FATAL("could not find target symbol for sym fixup for %s within %s",
                 target.str().c_str(), inSymbolId.str().c_str());
     }
+
     se->setSymbol(sym);
+    // Not all symExprs are noted as fixups (due to lowering and AST
+    // transformations), so visit the temporary converion symbol's recorded
+    // symExprs to try handle these stragglers.
+    for_SymbolSymExprs(se, tcsymbol) {
+      se->setSymbol(sym);
```

Comment:
It could be that `se->symbol()` is already fixed up to be `sym`. Should this code call `setSymbol` and add to fixedUp in this case? I think either way is OK. Not sure if there is any good reason for it to check or not check. But, I wanted to bring it up in case you think of something there.

Reply:
My understanding was that once `setSymbol` is called, the `se` would be removed from `tcsymbol`'s list; thus, it would not be discovered by `for_SymbolsSymExprs`. I got this from:
```C++
void SymExpr::setSymbol(Symbol* s)
{
  // If the old symbol is not NULL and the SymExpr
  // is in the tree, remove the SymExpr from the old Symbol's list.
  if (var != NULL && parentSymbol != NULL) {
    var->removeSymExpr(this);
  }
```

Reply:
Right, I was asking if it should skip calling `fixedUp.insert(se);` if `se->symbol() == sym`.

Reply:
Sorry, I am not understanding how `fixedUp.insert(se)` can be called (via `for_SymbolSymExprs`) if `sym == se->symbol() != tcsymbol`.

Reply:
Yes, now that you've pointed it out, I agree that it's not possible for `se->symbol() == tcsymbol` (which should be the case because you are using for_SymbolSymExprs) and also `se->symbol() == sym`. So, nothing to do here.